### ISSUE TITLE
Is `QuantileHistMaker` used anywhere?

### ISSUE THREAD

**Issue by trivialfis** [dmlc/xgboost#3755]
https://github.com/dmlc/xgboost/blob/9e73087324f72d8695e54da96e5e133769e662ca/src/tree/updater_histmaker.cc#L751

Doing a simple grep didn't find any usage of this class, is it obsoleted?

**Comment by hcho3** [comment]
Yes, I think it became obsolete. Commits back in year 2014 refer to it, but none afterwards. A PR to remove it would be great.
